src/main/java/com/epam/aidial/core/upstream/LoadBalancerProvider.java

```diff
@@ -0,0 +1,36 @@
+package com.epam.aidial.core.upstream;
+
+import com.epam.aidial.core.config.Upstream;
+import com.google.common.cache.Cache;
+import com.google.common.cache.CacheBuilder;
+
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+public class LoadBalancerProvider {
+
+    private final Cache<String, TieredBalancer> cache = CacheBuilder.newBuilder()
```

Comment:
The balancer state is not synchronised across Core instances. Should we keep it in the storage?

Reply:
As agreed with the team we decided not to keep the balancers' state across Core instances.